presto-hive/src/main/java/com/facebook/presto/hive/metastore/ColumnStatistics.java

```diff
@@ -0,0 +1,181 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.hive.metastore;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.Preconditions;
+
+import java.util.Optional;
+
+import static java.util.Arrays.stream;
+
+public class ColumnStatistics
+{
+    public enum Type
+    {
+        LONG,
+        DOUBLE,
+        DECIMAL,
+        BOOLEAN,
+        DATE,
+        STRING,
+        BINARY
+    }
+
+    private final Type type;
+    private final Optional<LongColumnStatistics> longColumnStatistics;
+    private final Optional<DoubleColumnStatistics> doubleColumnStatistics;
+    private final Optional<DecimalColumnStatistics> decimalColumnStatistics;
+    private final Optional<BooleanColumnStatistics> booleanColumnStatistics;
+    private final Optional<DateColumnStatistics> dateColumnStatistics;
+    private final Optional<StringColumnStatistics> stringColumnStatistics;
+    private final Optional<BinaryColumnStatistics> binaryColumnStatistics;
+
+    @JsonCreator
+    public ColumnStatistics(
```

Comment:
Make type part of constructor or move `getType` to `AbstractColumnStatistics` (if created)